src/AppInstallerCLICore/Workflows/UpdateFlow.h

```diff
@@ -54,12 +54,12 @@ namespace AppInstaller::CLI::Workflow
     // Outputs: None
     struct InstallOrUpgradeSinglePackage : public WorkflowTask
     {
-        InstallOrUpgradeSinglePackage(bool isUpgrade) :
-            WorkflowTask("InstallOrUpgradeSinglePackage"), m_isUpgrade(isUpgrade) {}
+        InstallOrUpgradeSinglePackage(SearchPurpose purpose) :
+            WorkflowTask("InstallOrUpgradeSinglePackage"), m_searchPurpose(purpose) {}

```

Comment:
We should check the value passed in is expected.

THROW_HR_IF(HRESULT_FROM_WIN32(ERROR_INVALID_STATE), m_searchPurpose != Install && m_searchPurpose != Upgrade);